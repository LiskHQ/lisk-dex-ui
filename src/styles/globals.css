:root {
  font-size: 16px;
}

html,
body,
#__next {
  padding: 0 !important;
  margin: 0;
  height: 100%;
}

a {
  color: inherit;
  text-decoration: none;
}

* {
  box-sizing: border-box;
}

a,
.btn,
button,
span,
p,
i,
input,
select,
textarea,
li,
img,
svg path,
*::after,
*::before,
.transition-3,
h1,
h2,
h3,
h4,
h5,
h6 {
  transition: all 0.3s ease-out 0s;
}

span,
p,
a,
li.MuiMenuItem-root,
label,
label.MuiFormLabel-root,
span.MuiTypography-root,
p.MuiTypography-root,
button.MuiButton-root,
button.MuiButtonBase-root,
.MuiInput-root,
.MuiTableCell-root {
  font-family: 'Roboto', sans-serif;
}

h1,
h2,
h3,
h4,
h5,
h6,
h1.MuiTypography-root,
h2.MuiTypography-root,
h3.MuiTypography-root,
h4.MuiTypography-root,
h5.MuiTypography-root,
h6.MuiTypography-root {
  font-family: 'Roboto', sans-serif;
}