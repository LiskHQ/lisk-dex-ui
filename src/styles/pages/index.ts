import { GovernanceStyle } from "./governance/index.style";

export {
  GovernanceStyle
}