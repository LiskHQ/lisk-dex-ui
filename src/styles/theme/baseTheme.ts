export const baseTheme = {
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 600,
    },
    h3: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 600,
    },
    h5: {
      fontSize: "1rem",
      lineHeight: "1.25rem",
      fontWeight: 500,
    },
    h6: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    body1: {
      fontSize: "1rem",
      lineHeight: "1.375rem",
      fontWeight: 500,
    },
    body2: {
      fontSize: "0.875rem",
      lineHeight: "1.125rem",
      fontWeight: 400,
    },
    caption: {
      fontSize: "0.875rem",
      lineHeight: "1rem",
      fontWeight: 500,
    },
    subtitle1: {
      fontSize: "1.75rem",
      lineHeight: "2rem",
      fontWeight: 500,
    },
    subtitle2: {
      fontSize: "1.25rem",
      lineHeight: "1.5rem",
      fontWeight: 400,
    }
  },
  icons: {
    sm: "1rem",
    md: "1.25rem",
    lg: "1.5rem",
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
    },
  },
  shadows: {
    sm: "0px 1px 3px rgba(0, 0, 0, 0.1)"
  }
}