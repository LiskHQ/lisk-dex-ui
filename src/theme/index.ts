export { default as theme } from "./theme";
export { default as variables } from "./variables";
