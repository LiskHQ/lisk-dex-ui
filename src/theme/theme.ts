import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
