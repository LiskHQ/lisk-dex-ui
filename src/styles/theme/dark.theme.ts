import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#F2F2F2',
    secondary: '#F2F2F2',
    paragraph: '#CDCDCD',
    body: '#F2F2F2',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#A4ABB7',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#5845CC',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#E6E6E6',
  },
  secondary: {
    0: '#F5981B'
  },
  primary: {
    85: '#7E6CF4',
    70: '#4F39B2',
    60: '#271B5B',
    40: '#2D2166',
    20: '#322573',
    10: '#4F39B2',
    5: '#322573',
    2.5: '#25195E',
    1: '#F4F8F9',
    0: '#FFFFFF',
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#E6E6E6',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#433199',
  },
  bg: {
    primary: 'linear-gradient(90deg, #140C42 0%, #211953 52.88%, #110A3A 101.53%, #140C42 101.53%)',
    secondary: '#22184C',
    helper: '#6953F4',
    proposalItem: 'linear-gradient(180deg, #2E2567 0%, rgba(37, 26, 103, 0.5) 143.01%);',
    banner: 'linear-gradient(180deg, #150556 0%, rgba(25, 9, 98, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #372589 0%, rgba(27, 19, 68, 0.8) 100%)',
    slider: '#322573',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#2EA3DD',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'dark',
  },
})

export default lightTheme
