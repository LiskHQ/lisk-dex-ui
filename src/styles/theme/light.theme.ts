import { createTheme } from '@mui/material/styles';
import { baseTheme } from './baseTheme';

const themeColors = {
  text: {
    primary: '#282828',
    secondary: '#515661',
    paragraph: '#6B7280',
    body: '#6B7280',
    heading: '#4738A6',
    button: '#FFFFFF',
    placeholder: '#6B7280',
  },
  lightcurve: {
    0: '#6953F4',
    10: '#A196F6',
    20: '#4738A6',
    40: '#362B7F',
    60: '#261E59',
    80: '#161133',
  },
  primary: {
    85: '#7E6CF4',
    60: '#A196F6',
    40: '#E4EBF9',
    20: '#D9D8F8',
    10: '#E7E9F9',
    5: '#EEF0F8',
    2.5: '#F2F5F9',
    1: '#F4F8F9',
    0: '#6953F4'
  },
  opacities: {
    5: '#333333',
    10: '#3D3D3D',
    20: '#535353',
    40: '#7E7E7E',
    80: '#D4D4D4',
  },
  border: {
    primary: '#E6E6E6',
  },
  bg: {
    primary: '#FBFBFB',
    secondary: '#FFFFFF',
    helper: '#22184D',
    banner: 'linear-gradient(180deg, #F3F3FD 0%, rgba(238, 238, 249, 0.9) 100%)',
    walletAddress: 'linear-gradient(180deg, #EEEFF7 0%, rgba(232, 234, 245, 0.8) 100%);',
    slider: '#F2F2F2',
  },
  success: {
    primary: '#E3F2E0',
    second: '#8DC881',
  },
  warning: {
    primary: '#F8F3D8',
    second: '#E2CA64',
  },
  info: {
    primary: '#EBF7F6',
    second: '#C0ECE8',
  },
  error: {
    primary: '#F6E4E6',
    second: '#FB8B96',
  },
} as const;

const lightTheme = createTheme({
  ...themeColors,
  ...baseTheme,
  palette: {
    mode: 'light'
  },
})

export default lightTheme
