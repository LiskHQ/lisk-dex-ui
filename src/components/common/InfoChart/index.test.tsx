import { ThemeProvider } from '@mui/material';
import { fireEvent, render } from '@testing-library/react';
import { IInfoChartProps, InfoChart } from './index';
import React from 'react';
import { lightTheme } from 'styles/theme';

function renderComponent(props: IInfoChartProps) {
  return render(
    <ThemeProvider theme={lightTheme}>
      <InfoChart {...props} />
    </ThemeProvider>
  );
}

describe('InfoChart component', () => {
  const mockProps: IInfoChartProps = {
    chartData: [
      {
        time: new Date(2023, 1, 1, 4, 50),
        price: 100,
      },
      {
        time: new Date(2023, 4, 1, 4, 50),
        price: 100,
      },
      {
        time: new Date(2023, 4, 28, 1, 40),
        price: 100,
      },
      {
        time: new Date(2023, 4, 29, 5, 30),
        price: 100,
      },
    ],
    tabs: ['Liquidity', 'TVL'],
    infoChartSummary: [
      {
        title: 'LSK Price',
        value: '$1.007',
        changePercent: 0.05,
      },
      {
        title: 'Total Liquidity',
        value: '$14.4m',
        changePercent: 2.32,
      },
    ],
    onTabChange: jest.fn(),
  };

  it('checks if the component matches the snapshot', () => {
    const { container } = renderComponent(mockProps);
    expect(container).toMatchSnapshot();
  });

  it('check period button works', () => {
    const { getByTestId } = renderComponent(mockProps);
    fireEvent.click(getByTestId('peroid-unit-D-test'));
    expect(getByTestId('volume-test')).toHaveTextContent('$100');

    fireEvent.click(getByTestId('peroid-unit-W-test'));
    expect(getByTestId('volume-test')).toHaveTextContent('$200');

    fireEvent.click(getByTestId('peroid-unit-M-test'));
    expect(getByTestId('volume-test')).toHaveTextContent('$300');

    fireEvent.click(getByTestId('peroid-unit-Y-test'));
    expect(getByTestId('volume-test')).toHaveTextContent('$400');
  });

  it('empty', () => {
    const { getByText } = renderComponent({
      ...mockProps,
      chartData: undefined
    });
    expect(getByText('Your favorite pools will be shown here.')).toBeInTheDocument();
  });

  it('check tab change work', () => {
    const { getByTestId } = renderComponent(mockProps);
    expect(getByTestId('tab-test-Liquidity')).toBeInTheDocument();
    fireEvent.click(getByTestId('tab-test-Liquidity'));
    expect(mockProps.onTabChange).toBeCalledWith('Liquidity');
  });
});

