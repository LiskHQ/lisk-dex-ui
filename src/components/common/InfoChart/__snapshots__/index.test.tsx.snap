// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoChart component checks if the component matches the snapshot 1`] = `
.emotion-0 {
  background: #FFFFFF;
}

.emotion-0 .info-chart-header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 1rem;
}

.emotion-0 .info-chart-header .info-chart-volume h2 {
  font-weight: 500;
  margin-top: 0.5rem;
  margin-bottom: 0.25rem;
}

.emotion-0 .info-chart-header .info-chart-tab {
  position: absolute;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  background: #E4EBF9;
  border: 0.5px solid #D9D8F8;
  border-radius: 0.75rem;
  padding: 0.125rem;
  min-height: 2rem;
}

.emotion-0 .info-chart-header .info-chart-tab .MuiTabs-scroller .MuiButtonBase-root {
  color: #6953F4;
  padding: 0.25rem 0.25rem;
  min-height: 1.5rem;
  border-bottom: none;
  text-transform: capitalize;
  z-index: 10;
}

.emotion-0 .info-chart-header .info-chart-tab .MuiTabs-scroller .MuiButtonBase-root.Mui-selected {
  color: #FFFFFF;
}

.emotion-0 .info-chart-header .info-chart-tab .MuiTabs-indicator {
  background: #6953F4;
  height: 100%;
  border-radius: 0.75rem;
}

.emotion-0 .info-chart-header .info-chart-period {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 .info-chart-header .info-chart-period .MuiButtonBase-root {
  padding: 0;
  min-width: 1.75rem!important;
  height: 1.25rem;
  margin-left: 0.25rem;
  box-shadow: none;
  background: #EEF0F8;
}

.emotion-0 .info-chart-header .info-chart-period .MuiButtonBase-root p {
  color: #6B7280;
}

.emotion-0 .info-chart-header .info-chart-period .MuiButtonBase-root.selected {
  background: #6953F4;
}

.emotion-0 .info-chart-header .info-chart-period .MuiButtonBase-root.selected p {
  color: #FFFFFF;
}

.emotion-0 .info-chart-timeline {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.75rem 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  border-top: 1px solid #E6E6E6;
  border-bottom: 1px solid #E6E6E6;
}

.emotion-0 .info-chart-summary {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 4.5rem 1rem 1rem 1rem;
}

.emotion-0 .info-chart-summary .summary-item .summary-title {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 0.25rem;
}

.emotion-0 .info-chart-summary .summary-item .summary-percent {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 0.25rem;
  color: #FF4557;
}

.emotion-0 .info-chart-summary .summary-item .summary-percent p {
  font-size: 0.75rem;
}

.emotion-0 .info-chart-summary .summary-item .summary-percent svg {
  margin-left: 0.25rem;
}

.emotion-0 .info-chart-summary .summary-item .summary-percent.increase {
  color: #459A33;
}

.emotion-0 .empty-chart-box {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 2rem;
  background: #FBFBFB;
  color: #6B7280;
}

.emotion-0 .empty-chart-box p {
  margin-top: 0.75rem;
}

.emotion-0 .empty-chart-box svg path {
  fill: #6B7280;
}

.emotion-3 {
  margin: 0;
  font-size: 1rem;
  line-height: 1.375rem;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
}

.emotion-4 {
  margin: 0;
  font-size: 2rem;
  font-weight: 600;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  line-height: 1.2;
  letter-spacing: -0.00833em;
}

.emotion-5 {
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.01071em;
}

.emotion-6 {
  overflow: hidden;
  min-height: 48px;
  -webkit-overflow-scrolling: touch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (max-width:575.95px) {
  .emotion-6 .MuiTabs-scrollButtons {
    display: none;
  }
}

.emotion-7 {
  position: relative;
  display: inline-block;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  white-space: nowrap;
  overflow-x: hidden;
  width: 100%;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.25;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  max-width: 360px;
  min-width: 90px;
  position: relative;
  min-height: 48px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  padding: 12px 16px;
  overflow: hidden;
  white-space: normal;
  text-align: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  color: rgba(0, 0, 0, 0.6);
}

.emotion-9::-moz-focus-inner {
  border-style: none;
}

.emotion-9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-9 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-9.Mui-selected {
  color: #1976d2;
}

.emotion-9.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-10 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-13 {
  position: absolute;
  height: 2px;
  bottom: 0;
  width: 100%;
  -webkit-transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: all 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  background-color: #1976d2;
}

.emotion-15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 16px;
  border-radius: 4px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #fff;
  background-color: #1976d2;
  box-shadow: 0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12);
  width: 100%;
  color: #FF4557;
  padding: 0.75rem;
  text-transform: capitalize;
}

.emotion-15::-moz-focus-inner {
  border-style: none;
}

.emotion-15.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-15 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-15:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #1565c0;
  box-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);
}

@media (hover: none) {
  .emotion-15:hover {
    background-color: #1976d2;
  }
}

.emotion-15:active {
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
}

.emotion-15.Mui-focusVisible {
  box-shadow: 0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12);
}

.emotion-15.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  box-shadow: none;
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-15.MuiButton-outlined {
  border-color: #FF4557;
}

.emotion-15.MuiButton-contained {
  background: #6953F4;
  color: #FFFFFF;
}

.emotion-15.Mui-disabled {
  opacity: 0.7;
}

.emotion-27 {
  width: 100%;
  height: 12.5rem;
}

.emotion-27 svg {
  width: 100%;
  height: 12.5rem;
}

.emotion-29 {
  margin: 0;
  font-size: 0.75rem;
  line-height: 1rem;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.03333em;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="info-chart-header MuiBox-root emotion-1"
    >
      <div
        class="info-chart-volume MuiBox-root emotion-1"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-3"
        >
          Volume
        </p>
        <h2
          class="MuiTypography-root MuiTypography-h2 emotion-4"
          data-testid="volume-test"
        >
          $
          100
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-5"
        >
          Tue May 30 2023
        </p>
      </div>
      <div
        class="MuiTabs-root info-chart-tab emotion-6"
      >
        <div
          class="MuiTabs-scroller MuiTabs-fixed emotion-7"
          style="overflow: hidden; margin-bottom: 0px;"
        >
          <div
            class="MuiTabs-flexContainer MuiTabs-centered emotion-8"
            role="tablist"
          >
            <button
              aria-selected="true"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected emotion-9"
              role="tab"
              tabindex="0"
              type="button"
            >
              Liquidity
              <span
                class="MuiTouchRipple-root emotion-10"
              />
            </button>
            <button
              aria-selected="false"
              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary emotion-9"
              role="tab"
              tabindex="-1"
              type="button"
            >
              TVL
              <span
                class="MuiTouchRipple-root emotion-10"
              />
            </button>
          </div>
          <span
            class="MuiTabs-indicator emotion-13"
            style="left: 0px; width: 0px;"
          />
        </div>
      </div>
      <div
        class="info-chart-period MuiBox-root emotion-1"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium selected emotion-15"
          data-testid="peroid-unit-D-test"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            D
          </p>
          <span
            class="MuiTouchRipple-root emotion-10"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium  emotion-15"
          data-testid="peroid-unit-W-test"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            W
          </p>
          <span
            class="MuiTouchRipple-root emotion-10"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium  emotion-15"
          data-testid="peroid-unit-M-test"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            M
          </p>
          <span
            class="MuiTouchRipple-root emotion-10"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium  emotion-15"
          data-testid="peroid-unit-Y-test"
          tabindex="0"
          type="button"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            Y
          </p>
          <span
            class="MuiTouchRipple-root emotion-10"
          />
        </button>
      </div>
    </div>
    <div
      class="info-chart-svg emotion-27"
    >
      <svg />
    </div>
    <div
      class="info-chart-timeline MuiBox-root emotion-1"
    >
      <span
        class="MuiTypography-root MuiTypography-caption emotion-29"
      >
        5:30 AM
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-29"
      >
        5:30 AM
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-29"
      >
        5:30 AM
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-29"
      >
        5:30 AM
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-29"
      >
        5:30 AM
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-29"
      >
        5:30 AM
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-29"
      >
        5:30 AM
      </span>
    </div>
    <div
      class="info-chart-summary MuiBox-root emotion-1"
    >
      <div
        class="summary-item MuiBox-root emotion-1"
      >
        <div
          class="summary-title MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-3"
          >
            LSK Price
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-5"
        >
          $1.007
        </p>
        <div
          class="summary-percent MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            -2.34
            %
          </p>
          <svg
            fill="none"
            height="12"
            viewBox="0 0 12 12"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 9L6.75 4.25L4.25 6.75L0.5 3"
              stroke="#FF4557"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M8.5 9H11.5V6"
              stroke="#FF4557"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
      <div
        class="summary-item MuiBox-root emotion-1"
      >
        <div
          class="summary-title MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-3"
          >
            Total Liquidity
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-5"
        >
          $14.4m
        </p>
        <div
          class="summary-percent increase MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            2.32
            %
          </p>
          <svg
            fill="none"
            height="12"
            viewBox="0 0 12 12"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 3L6.75 7.75L4.25 5.25L0.5 9"
              stroke="#459A33"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M8.5 3H11.5V6"
              stroke="#459A33"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
      <div
        class="summary-item MuiBox-root emotion-1"
      >
        <div
          class="summary-title MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-3"
          >
            Volume 24h
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-5"
        >
          $2.5m
        </p>
        <div
          class="summary-percent increase MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            1.45
            %
          </p>
          <svg
            fill="none"
            height="12"
            viewBox="0 0 12 12"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 3L6.75 7.75L4.25 5.25L0.5 9"
              stroke="#459A33"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M8.5 3H11.5V6"
              stroke="#459A33"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
      <div
        class="summary-item MuiBox-root emotion-1"
      >
        <div
          class="summary-title MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-3"
          >
            Fees 24h
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-5"
        >
          $48.9k
        </p>
        <div
          class="summary-percent increase MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-5"
          >
            4.86
            %
          </p>
          <svg
            fill="none"
            height="12"
            viewBox="0 0 12 12"
            width="12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.5 3L6.75 7.75L4.25 5.25L0.5 9"
              stroke="#459A33"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
            <path
              d="M8.5 3H11.5V6"
              stroke="#459A33"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
      </div>
      <div
        class="summary-item MuiBox-root emotion-1"
      >
        <div
          class="summary-title MuiBox-root emotion-1"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 emotion-3"
          >
            Transactions 24h
          </p>
        </div>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-5"
        >
          621
        </p>
      </div>
    </div>
  </div>
</div>
`;
